internal/worker/firewaller/firewaller.go

```diff
@@ -937,8 +947,7 @@ func (fw *Firewaller) ingressRulesForExposedMachineUnit(ctx context.Context, uni
 		// access the port ranges opened for this endpoint; then resolve
 		// the CIDRs for the spaces specified in the expose details to
 		// construct the full source CIDR list for the generated rules.
-		srcCIDRs := set.NewStrings(exposeDetails.ExposeToCIDRs...)
-		for _, spaceID := range exposeDetails.ExposeToSpaces {
+		for _, spaceID := range exposeDetails.ExposeToSpaceIDs.Values() {
```

Comment:
Move this for loop into a method, this method is getting way too big now.

Reply:
Done, see 3447f80d5d326b630a8e23bb3e5f27d93ae88212.